src/plonk/lookup/prover.rs

```diff
@@ -327,35 +330,42 @@ impl<C: CurveAffine> Permuted<C> {
         // Compute the evaluations of the lookup product polynomial
         // over our domain, starting with z[0] = 1
         let z = iter::once(C::Scalar::one())
-            .chain(lookup_product.into_iter().skip(1))
+            .chain(lookup_product)
```

Comment:
We talked about this in a pairing.

This witness generation code needs to produce `u` values for `z`; given that z[0] = 1, we therefore need u - 1 lookup products to finish populating `z`. We don't need to produce the `u + 1` value because we don't need it.

Reply:
Also, where we were doing Z(X) previously, we are now doing Z(ωX).